website/src/models/group.js

```diff
@@ -352,6 +354,77 @@ GroupSchema.methods.toJSON = function() {
   return doc;
 };
 
+GroupSchema.statics.removeUserFromChallenges = function(user, group, keep, cb) {
```

Comment:
Plus, if something new comes along that a user needs to be removed from, all we need to do to update the `removeUserFromAll` method is to create a new private method and tack it on to the end, rather than having to parse out exactly what is happening in the rest of the method to make sure we're doing it right. 

Also, creating the method like this eliminates the need for most of your comments, since the method name tells you what is happening.